numba/cuda/compiler.py

```diff
@@ -261,6 +261,9 @@ def __init__(self, kernel, ntasks, tpb, stream, sharedmem):
         self.sharedmem = sharedmem
 
     def __call__(self, *args):
+        if self.ntasks == 0:
+            return
```

Comment:
Should this also handle `.ntasks < 0` and raise, else it'll just end up as an error at kernel launch?

Reply:
Yes, it probably should. Will add this.

Reply:
I've decided to guard for this in the initialiser instead - it seems to me that it's legitimate for a `ForAll` to have no work to do and therefor the call to it be a no-op, but it's not legitimate to even create a `ForAll` with negative work.